daemons/controld/controld_schedulerd.c

```diff
@@ -438,11 +438,20 @@ force_local_option(xmlNode *xml, const char *attr_name, const char *attr_value)
     freeXpathObject(xpathObj);
 }
 
+static gboolean
+sleep_timer(gpointer data)
+{
+    controld_set_fsa_action_flags(A_PE_INVOKE);
+    controld_trigger_fsa();
+    return G_SOURCE_CONTINUE;
+}
```

Comment:
Since this is just replacing the sleep and following code in `do_pe_invoke_callback`, I think this is supposed to be a one-off timer.  That is, it should only execute one time and the way we do that is by returning `G_SOURCE_REMOVE` and passing `FALSE` to `mainloop_timer_add`.